tiledb/sm/cpp_api/array.h

```diff
@@ -1252,8 +1256,14 @@ class Array {
       const std::string& uri,
       tiledb_encryption_type_t encryption_type,
       const std::string& encryption_key,
-      Config* const config = nullptr) {
```

Comment:
Isn't this an API breakage? Perhaps just assign to a local `Config` copy and pass that local config to the function below?

Reply:
Good question: my answer depends on how we define "breakage". This change is less restrictive so it will not break existing code. However, it is _different_.

Reply:
Wait sorry, I have that backwards. I'll add a local copy and revert this const.